course_access_groups/views.py

```diff
@@ -5,20 +5,55 @@
 
 from __future__ import absolute_import, unicode_literals
 
+
 from rest_framework import viewsets
 from rest_framework.pagination import LimitOffsetPagination
 
-
-from course_access_groups.serializers import GroupSerializer
+from course_access_groups.serializers import (
+    GroupSerializer,
+    MembershipSerializer,
+    RuleSerializer,
+    CourseGroupSerializer,
+)
 from course_access_groups.models import (
-    CourseAccessGroup,
+    Group,
+    Membership,
+    Rule,
+    CourseGroup,
 )
 
 
-class GroupsViewSet(viewsets.ModelViewSet):
-    model = CourseAccessGroup
+class GroupViewSet(viewsets.ModelViewSet):
```

Comment:
Since the serializers appear to be read only, you know you can just make the view inherit from `ReadOnlyModelViewSet`? Because who really cares about making sure data is read only, the view or the serializer? I ask because it looks like all fields in the serializers are being explicitly set as read only

See here: https://www.django-rest-framework.org/api-guide/viewsets/#readonlymodelviewset

Reply:
They're not readonly. Just the related fields e.g.  `_name` are read only.